### ISSUE TITLE
Add GeoJSON support for geo_point field value formats

### ISSUE THREAD

**Issue by talevy** [elastic/elasticsearch#47815]
**Proposal**

Support GeoJSON's `Point` type.

**Existing Support**

`geo_point` currently supports the following formats

- `"POINT(0 0)"`, WKT
- `{ "lat": 0, "lon": 0}`, lat/lon object
- `[0, 0]`, [lon, lat] array
- `"0, 0"`, (lat, lon) string
- `"gbsuv"`, geohash

**Reason**

Adding support for GeoJSON would make it easier to index the same field value as both geo_point and geo_shape when GeoJSON is preferred over WKT.

**Comment by nknize** [comment]
@talevy is this related to #47535?

**Comment by nknize** [comment]
> @talevy is this related to #47535?

nvm.. I just saw it in the comments.

**Comment by talevy** [comment]
yeah... it took me a second to realize these are slightly different issues, where the other issue would supersede this one if done

**Comment by imotov** [comment]
@talevy do we still want to do that?

**Comment by talevy** [comment]
I, personally, do not see why this should not be supported. That being said, the silence on this issue reflects a lack of demand for this feature...

**Comment by imotov** [comment]
I think it should help consistency. I will keep it open for now, but we can discuss it in a team meeting if needed.

**Comment by jtibshirani** [comment]
For consistency it would be great to add support for GeoJSON when indexing `point` fields as well.

**Comment by craigtaverner** [comment]
This work is related to https://github.com/elastic/elasticsearch/issues/84599 and there is now even a PR at https://github.com/elastic/elasticsearch/pull/85120
